gh-136752: Clarify documentation for ``IPv{N}Address.is_reserved`` (#136794)
python/cpython: Doc/library/ipaddress.rst

   .. attribute:: is_reserved

>       ``True`` if the address is *labelled* as reserved by the IETF:
> 
>       * IPv4 address ``240.0.0.0/4`` named ``Reserved``. This attribute is not related to the
>         ``Reserved-by-Protocol`` value in iana-ipv4-special-registry_
>       * IPv6 addresses allocated as ``Reserved by IETF`` for future use (see iana-ipv6-address-space_)

Reviewer: ```suggestion
      ``True`` if the address is noted as reserved by the IETF.
      For IPv4, this is only ``240.0.0.0/4``, the "Reserved" address block.
      For IPv6, this is all addresses `allocated <iana-ipv6-address-space_>`__ as
      ``Reserved by IETF`` for future use.

      Note: ``is_reserved`` is not related to the "Reserved-by-Protocol" value.

      .. seealso::

         iana-ipv4-special-registry_
```

   .. attribute:: is_reserved

      ``True`` if the address is noted as reserved by the IETF.
      For IPv4, this is only ``240.0.0.0/4``, the ``Reserved`` address block.
      For IPv6, this is all addresses `allocated <iana-ipv6-address-space_>`__ as
      ``Reserved by IETF`` for future use.

      .. note:: For IPv4, ``is_reserved`` is not related to the address block value of the
        ``Reserved-by-Protocol`` column in iana-ipv4-special-registry_.

      .. caution:: For IPv6, ``fec0::/10`` a former Site-Local scoped address prefix is

>          currently excluded from that list (see :attr:`~IPv6Address.is_site_local`).

Reviewer: ```suggestion
         currently excluded from that list (see :attr:`~IPv6Address.is_site_local` & :rfc:`3879`).
```
